apps/server/src/modules/tool/controller/dto/request/lti11-tool-config.params.ts

```diff
@@ -0,0 +1,36 @@
+import { ExternalToolConfigCreateParams } from '@src/modules/tool/controller/dto/request/external-tool-config.params';
+import { LtiMessageType } from '@src/modules/tool/interface/lti-message-type.enum';
+import { LtiRole } from '@src/modules/tool/interface/lti-role.enum';
+import { ToolConfigType } from '@src/modules/tool/interface/tool-config-type.enum';
+import { IsEnum, IsString } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class Lti11ToolConfigParams extends ExternalToolConfigCreateParams {
+	@IsString()
+	@ApiProperty()
+	key!: string;
+
+	@IsString()
+	@ApiProperty()
+	secret!: string;
+
+	@IsString()
+	@ApiProperty()
+	resource_link?: string;
+
+	@IsEnum(LtiMessageType)
+	@ApiProperty()
+	lti_message_type!: LtiMessageType;
+
+	@IsEnum(LtiRole)
+	@ApiProperty()
+	roles!: LtiRole[];
```

Comment:
idk if alex told you but roles, launch_presentation_locale and launch_presentation_document_target will not be part of the lti11 config anymore, but will instead be set via the custom params (to save complexity and development time).